utils/llm.py

```diff
@@ -146,24 +114,21 @@ def make_assistant_message(content: str) -> LLMMessage:
     return make_message("assistant", content)
 
 
-def context_token_len(context: LLMContext, model: Optional[str] = None, response: bool = False) -> int:
-    # Extracted from here:
-    # https://github.com/openai/openai-cookbook/blob/main/examples/How_to_count_tokens_with_tiktoken.ipynb
-    encoder = tiktoken.encoding_for_model(get_model(model))
-    num_tokens = 0
-    for message in context:
-        num_tokens += 4  # every message follows <|start|>{role/name}\n{content}<|end|>\n
-        num_tokens += len(encoder.encode(message["content"]))
-    if response:
-        num_tokens += 3  # every reply is primed with <|start|>assistant<|message|>
-    return num_tokens
+def context_token_len(context: LLMContext, model: str = CHEAPEST_TOKENISER) -> int:
+    model = model_from_alias(model)
+    return litellm.token_counter(model, messages=context)
 
 
-def get_tokens_for_script(script: list[str], model: Optional[str] = None) -> int:
-    encoder = tiktoken.encoding_for_model(get_model(model))
+def tokens_in_script(script: list[str], model: str = CHEAPEST_TOKENISER) -> int:
+    model = model_from_alias(model)
     num_tokens = 0
     for line in script:
         num_tokens += 4
-        num_tokens += len(encoder.encode(line))
+        num_tokens += litellm.token_counter(model, text=line)
```

Comment:
The `+= 4` is inherited from the old `context_token_len`, which was intended for use with OpenAI models (taken [from here](https://github.com/openai/openai-cookbook/blob/main/examples/)). I'd add a comment or something, so that we don't forget why that number is there e.g. `# every message follows <|start|>{role/name}\n{content}<|end|>\n`